programs/drift/src/controller/orders.rs

```diff
@@ -2982,40 +2985,34 @@ pub fn place_spot_order(
         )?;
     }
 
-    let (worst_case_token_amount_after, _) = user.spot_positions[spot_position_index]
-        .get_worst_case_token_amount(spot_market, &oracle_price_data, None, None)?;
-
-    let order_direction = user.orders[new_order_index].direction;
-    let base_asset_amount_unfilled = user.orders[new_order_index]
-        .get_base_asset_amount_unfilled(Some(signed_token_amount.cast()?))?;
-    let order_risk_decreasing = is_order_risk_decreasing(
-        &order_direction,
-        base_asset_amount_unfilled,
-        signed_token_amount.cast()?,
+    let OrderFillSimulation {
+        free_collateral_contribution: free_collateral_contribution_after,
+        ..
+    } = user.spot_positions[spot_position_index].get_worst_case_token_amount(
+        spot_market,
+        &strict_oracle_price,
+        Some(signed_token_amount),
+        MarginRequirementType::Initial,
     )?;
 
     // Order fails if it's risk increasing and it brings the user collateral below the margin requirement
-    let risk_decreasing = worst_case_token_amount_after.unsigned_abs()
-        <= worst_case_token_amount_before.unsigned_abs()
-        && order_risk_decreasing;
+    let risk_increasing = free_collateral_contribution_after < free_collateral_contribution_before;
```

Comment:
is it dumb to just have a function `is_risk_increasing(free_collateral_contribution_before, free_collateral_contribution_after)` since i've seen this line more than once?

Reply:
nah, can do that